### PR TITLE
CNF-14303: Initial drop of test plugin

### PR DESCRIPTION
**Pull request by donpenney**
This PR is the initial drop of the Test Hardware Manager Plugin. It uses a configmap for managing a set of fake node resources, processing NodePool CRs and allocating nodes from the configmap. See README for more details.

### CODE REVIEW

**Comment by donpenney** [comment]
The PR is pretty big, primarily due to the vendor folder. It might be a little more manageable/responsive if I just add a commit that removes the vendor folder and put it back after this PR is merged

EDIT: Yeah, that dropped close to 4k files from the PR. Definitely more manageable

**Comment by donpenney** [comment]
The first couple of commits are setting up the initial framework with generated code, using operator-sdk commands. I've included the commands run in the commit messages for those commits, for reproducibility / as a template for creating new plugins

File: .github/workflows/requirements.txt [hunk 1]
```diff
@@ -0,0 +1,17 @@
+#
+# Copyright (c) 2023 Red Hat, Inc.
```
Comment (bartwensley) [inline .github/workflows/requirements.txt]: nit: update the year

File: README.md [hunk 2]
```diff
@@ -0,0 +1,205 @@
...
+  namespace: oran-hwmgr-plugin-test
+  resourceVersion: "16208107"
+  uid: 0838467c-f73e-4c5b-86cf-0083352243ce
+spec:
+  groupName: master
+  hwProfile: profile-spr-single-processor-64G
+  nodePool: testcloud-1
+status:
+  bmc:
+    address: idrac-virtualmedia+https://192.168.2.1/redfish/v1/Systems/System.Embedded.1
```
Comment (bartwensley) [inline README.md:114]: nit: Assuming the IP addresses aren't used for anything, you should probably update the tool and examples to use the documentation ranges (e.g. 192.0.2.0/24).

File: README.md [hunk 3]
```diff
@@ -0,0 +1,205 @@
...
+    credentialsName: bmcSecret-dummy-sp-64g-1
+  bootMACAddress: c6:b6:13:a0:02:01
```
Comment (bartwensley) [inline README.md:116]: nit: could make this more obvious as a fake MAC with something like: 0A:1B:2C:3D:4E:5F

File: bundle.Dockerfile [hunk 4]
```diff
@@ -0,0 +1,20 @@
+FROM scratch
+
+# Core bundle labels.
+LABEL operators.operatorframework.io.bundle.mediatype.v1=registry+v1
+LABEL operators.operatorframework.io.bundle.manifests.v1=manifests/
+LABEL operators.operatorframework.io.bundle.metadata.v1=metadata/
+LABEL operators.operatorframework.io.bundle.package.v1=oran-hwmgr-plugin-test
+LABEL operators.operatorframework.io.bundle.channels.v1=alpha
+LABEL operators.operatorframework.io.metrics.builder=operator-sdk-v1.35.0
```
Comment (bartwensley) [inline bundle.Dockerfile:9]: Don't know if it matters, but the o2-ims operator seems to be using version 1.36.1 - did you want to use the latest version?

**Review by bartwensley — COMMENTED** [review]
I took a quick look and have a few minor comments/questions, but you really need @tliu2021 to take a better look.

File: .github/workflows/requirements.txt (hunk 1 above)
Comment (donpenney) [reply .github/workflows/requirements.txt]: will do... copied from o2ims repo

File: README.md (hunk 2 above)
Comment (donpenney) [reply README.md:114]: I was just creating dummy values out of the private subnet range, using the last two values to map to indexes of profile/node for uniqueness

File: README.md (hunk 3 above)
Comment (donpenney) [reply README.md:116]: I'm not sure where I got the "c6:b6:13" from now, but I thought it was a "private" mac range. From what I can see, the "6" in the "c6" denotes the mac as being private. From https://en.wikipedia.org/wiki/MAC_address, it looks like "x6" would indicate a "reserved" locally administered address. And like the ip, I'm using the last two fields as indexes into profile/node.

File: bundle.Dockerfile (hunk 4 above)
Comment (donpenney) [reply bundle.Dockerfile:9]: 1.35.0 was the latest when I initially installed the operator-sdk to start on this. 1.36.1 is fairly new

**Comment by tliu2021** [comment]
/lgtm
Good stuff!

**Pull request merged**